text-minimessage/src/main/java/net/kyori/adventure/text/minimessage/internal/TagInternals.java

```diff
@@ -36,7 +36,8 @@
  */
 @ApiStatus.Internal
 public final class TagInternals {
-  private static final Pattern TAG_NAME_PATTERN = Pattern.compile("[!?#]?[a-z0-9_-]*");
+  public static final String TAG_NAME = "[!?#]?[a-z0-9_-]*";
```

Comment:
Should be given a better name, perhaps `TAG_NAME_REGEX`? Also should probably be documented.